src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -953,29 +969,41 @@ private boolean checkForSaveAndContinue() {
     return willContinue;
   }
 
-  private boolean checkForSaveAndExecute() {
-    boolean willContinue = false;
-    if (checkIfNeedsSaving()) {
+  private String commitConfigToFileAndGetFilepath() {
+    String filename = null;
+    ConfigComparisonResult comparisonResult = compareConfigs();
+    if (comparisonResult != ConfigComparisonResult.SAME) {
       ButtonType saveButton = new ButtonType("Save", ButtonBar.ButtonData.YES);
+      ButtonType dontSaveButton = new ButtonType("Proceed without Saving", ButtonBar.ButtonData.NO);
+      ButtonType cancelButton = new ButtonType("Cancel", ButtonBar.ButtonData.NO);
+      ButtonType noButton = comparisonResult
+          == ConfigComparisonResult.DIFFERENT_BUT_VERSION_IS_TEST ? dontSaveButton : cancelButton;
       Alert alert =
           new Alert(
               AlertType.WARNING,
               "You must either save your changes before continuing or load a new contest config!",
```

Comment:
This message also appears when using test files, which is a little misleading now that it's possible to bypass it. Maybe we make the message say this for tests:

"You are using a test config, and must either save the file or proceed with creating a temporary one!"

Reply:
And then maybe we also change the "Proceed without Saving" button to "Proceed with Temporary File".